plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -775,6 +775,14 @@ func RunGeneralPredicates(pod *api.Pod, nodeName string, nodeInfo *schedulercach
 	if !fit {
 		return fit, err
 	}
+	// TODO(HaiyangDING): Make a proper fix for it.
```

Comment:
I think the comment here should instead be exactly the same comment as the other place where we check max pods, namely

// TODO: move the following podNumber check to podFitsResourcesInternal when Kubelet allows podNumber check (See #20263).